File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.py
class Solution:
    def closeStrings(self, word1: str, word2: str) -> bool:


        cntr1 = Counter(word1) 
        cntr2 = Counter(word2) 
        
        bool1 = set(cntr1.keys()) == set(cntr2.keys())
        bool2 =  sorted(list(cntr1.values())) == sorted(list(cntr2.values())) 
        return bool1 & bool2 

File: 1657-determine-if-two-strings-are-close/1657-determine-if-two-strings-are-close.cpp
class Solution {
public:
    bool closeStrings(string word1, string word2) {
        
       vector<int> cntr1(26,0) , cntr2(26,0) ;
       set<char> set1 , set2 ; 
        for(char e: word1){
            cntr1[e-'a']++;
            set1.insert(e) ; 
        }
        for(char e: word2){
            cntr2[e-'a']++ ; 
            set2.insert(e); 
        }
        
        sort(begin(cntr1) , end(cntr1)); 
        sort(begin(cntr2) , end(cntr2)); 
        
        return cntr1 == cntr2 && set1 == set2 ; 
        
    }
};